common/src/primitives/height.rs

```diff
@@ -1,18 +1,113 @@
 use parity_scale_codec_derive::{Decode, Encode};
+use std::fmt;
+use std::ops::{Add, Sub};
+
+type HeightIntType = u64;
+type DistanceIntType = i64;
 
 #[derive(Debug, Copy, Clone, PartialOrd, Ord, PartialEq, Eq, Encode, Decode)]
-pub struct BlockHeight(u64);
+pub struct BlockHeight(HeightIntType);
+
+// Display should be defined for thiserr crate
+impl fmt::Display for BlockHeight {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.fmt(f)
+    }
+}
+
+impl From<BlockHeight> for HeightIntType {
+    fn from(block_height: BlockHeight) -> HeightIntType {
+        block_height.0
+    }
+}
+
+impl From<HeightIntType> for BlockHeight {
+    fn from(w: HeightIntType) -> BlockHeight {
+        BlockHeight(w)
+    }
+}
+
+impl Add<BlockDistance> for BlockHeight {
+    type Output = Option<BlockHeight>;
+
+    fn add(self, other: BlockDistance) -> Option<BlockHeight> {
+        let height: i64 = self.0.try_into().ok()?;
```

Comment:
There are many places in this file where `i64` or `u64` is used. It seems to me that often, the `DistanceIntType` or `HeightIntType` respectively would be more appropriate.

Reply:
I'd rather keep them like this because I'm always concerned if these change how everything else will be affected. If these types are fixed, no changes of the types will break the code.